decoder.go

```diff
@@ -167,59 +247,50 @@ func (d *Decoder) Read(p []byte) (int, error) {
 		// Move any untransformed source bytes to the start of the buffer
 		// and read more bytes.
 		if d.src0 != 0 {
+			d.clampSrc()
 			d.src0, d.src1 = 0, copy(d.src, d.src[d.src0:d.src1])
 		}
 		n, d.err = d.r.Read(d.src[d.src1:])
 		d.src1 += n
 	}
-}
+} // end func Read
 
 // Transform starts by reading line-by-line to parse yEnc headers until it encounters yEnc data.
 // It then incrementally decodes chunks of yEnc encoded data before returning to line-by-line processing
 // for the footer and EOF pattern (.\r\n)
 func (d *Decoder) Transform(dst, src []byte, atEOF bool) (nDst, nSrc int, err error) {
 transform:
-	if d.body && d.format == FormatYenc {
-		nd, ns, end, _ := DecodeIncremental(dst[nDst:], src[nSrc:], &d.State)
-		if nd > 0 {
-			d.hash.Write(dst[nDst : nDst+nd])
-			d.actualSize += int64(nd)
-			nDst += nd
-		}
-
-		switch end {
-		case EndControl:
-			nSrc += ns - 2
-			d.body = false
-		case EndArticle:
-			nSrc += ns - 3
-			d.body = false
-		default:
-			if d.State == StateCRLFEQ {
-				d.State = StateCRLF
-				nSrc += ns - 1
-			} else {
-				nSrc += ns
-			}
-			return nDst, nSrc, transform.ErrShortSrc
-		}
-	}
-
```

Comment:
Moving this drastically reduced speed because it ends up processing line-by-line.

See `go test -bench=.`

I accept that the logic flow is a bit funky with a goto, I believe that was mainly because once it reads the `=ypart` we know the rest of the buffer is body so I wanted to process it rather than having to wait for the next Transform call. It could probably be refactored to remove the gotos.

This is related to a comment below which is also processing yenc line-by-line.